src/pages/blogDetailPage/BlogDetail.jsx

```diff
@@ -0,0 +1,82 @@
+import {useEffect, useState} from "react";
+import { Link, useParams, useNavigate } from "react-router-dom";
+import axios from "axios";
+import { CaretLeft, Clock } from "@phosphor-icons/react";
+import Button from "../../components/button/Button.jsx";
+import ErrorMessage from "../../components/errorMessage/ErrorMessage.jsx";
+import formatDateString from "../../helpers/formatDateString.js";
+import './BlogDetail.css';
+
+
+function BlogDetail() {
+    const { id } = useParams();
+    const navigate = useNavigate();
+    const [postDetails, setPostDetails] = useState({});
```

Comment:
Ik zou zelf als default waarde `undefined` nemen. Je check verderop wordt daarmee een stukje simpeler
```suggestion
    const [postDetails, setPostDetails] = useState();
```